app/src/main/java/ai/elimu/analytics/task/ExportEventsToCsvWorker.kt

```diff
@@ -164,12 +165,9 @@ class ExportEventsToCsvWorker(context: Context, workerParams: WorkerParameters)
 
                 // Write the content to the CSV file
                 val filesDir = applicationContext.filesDir
-                val versionCodeDir = File(
-                    filesDir,
-                    "version-code-$versionCode"
-                )
-                val letterSoundLearningEventsDir =
-                    File(versionCodeDir, "letter-sound-learning-events")
+                val language = SharedPreferencesHelper.getLanguage(applicationContext)
+                val languageDir = File(filesDir, "lang-${language}")
+                val letterSoundLearningEventsDir = File(languageDir, "letter-sound-learning-events")
```

Comment:
This logic block repeats itself for every event.
We should extract it somewhere as a function and re-use the logic to avoid code duplication and cost of maintenance in the future.
Also, take this chance to stop using hardcoded event type. 
Instead, use predefined `LearningEventUploadType` enums.

You can refer to https://github.com/elimu-ai/analytics/pull/295 in file: app/src/main/java/ai/elimu/analytics/task/ExportEventsToCsvWorker.kt for an example.

Reply:
@tuancoltech Made an issue for handling refactoring separately: 
* https://github.com/elimu-ai/analytics/issues/303